packages/wrangler/src/config/validation-helpers.ts

```diff
@@ -550,7 +550,7 @@ export const getBindingNames = (value: unknown): string[] => {
 	} else if (isNamespaceList(value)) {
 		return value.map(({ binding }) => binding);
 	} else if (isRecord(value)) {
-		return Object.keys(value);
+		return Object.keys(value).filter((k) => k !== "bindings");
```

Comment:
There is some weirdness here.

In some tests where it passes a KV binding with no names and no unsafe bindings at all, the if (isBinding.. on line 548 doesn't match.  We end up in this branch and for both cases add the name "bindings" as if it were a binding (its not, its the bindings array).

Reply:
I agree this is not right but I think the correct approach is to filter out properties that are undefined, since they are clearly not bindings.